Allow smooth migration for failed transactions column

bartekn opened the pull request:
Migration file added in https://github.com/stellar/go/pull/756 made it impossible to migrate database smoothly:
```sql
ALTER TABLE history_ledgers RENAME column transaction_count TO successful_transaction_count;
```
because old version could not work with new schema. This PR adds a new column (leaving the existing `transaction_count`) and adds a new code path to support displaying new data if it's there.

bartekn commented on services/horizon/internal/actions_trade_test.go:
```diff
@@ -392,8 +392,8 @@ func TestTradeActions_SyntheticOfferIds(t *testing.T) {
-			assertOfferType(ht, records[0].BaseOfferID, TOIDType)
-			assertOfferType(ht, records[1].BaseOfferID, TOIDType)
+			assertOfferType(ht, records[0].CounterOfferID, TOIDType)
+			assertOfferType(ht, records[1].CounterOfferID, TOIDType)
```
@tomerweller can you take a look if we can make this deterministic. It seems we need to change from time to time when tests are rebuilt.

tomerweller replied:
Assets get an incremental id when they are first encountered in ingestion. The base//counter canonical form is driven by those ids (base is lower, counter is higher). When we change ingestion order, or the scenarios tx order, or stellar-core tx ordering changes, these ids can change and the canonical order can change. 

This also has the unpleasant effect of ids being different from one horizon instance to another. Though, I don't think we ever expose these ids externally. 

We can, alternatively use a deterministic issuer+assettype+assetcode id, but that means much more data on the id column. Might be worth it though?

Anyway, i wouldn't make any changes in the context of this PR.

howardtw replied:
@bartekn do you want to open an issue for this?

bartekn replied:
https://github.com/stellar/go/issues/850

tomerweller commented:
@howardtw regarding base/counter, see my comment here: https://github.com/stellar/go/pull/847/files/9613be790081b9fc4b1e1e2a406761a293002d1c#r252775698

bartekn commented:
Implemented the changed requested. Please take a look!

> I'm not sure why we change BaseOfferID to CounterOfferID in this PR.

https://github.com/stellar/go/pull/847#issuecomment-459482950

> Do we have to rebuild the test scenario because we want to make sure the ingestion works? If so, do we also want to test the read from the new column?

Added a simple test to check this.
